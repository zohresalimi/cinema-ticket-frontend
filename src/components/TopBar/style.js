import styled from "styled-components";

export const Div = styled.div`
  flex-direction: row;
  justify-content: inherit;
  align-items: center;
  display: flex;
  height: 100%;
`;

export const TopMenuContainer = styled.div`
  font-family: "sfMedium";
  display: flex;
  align-items: center;
  width: 100%;
  height: 4rem;
  background-color: ${(props) => props.theme.colors.red};
  color: ${(props) => props.theme.colors.white};
`;

export const Wrapper = styled.div`
  top: 0;
  left: 0;
  width: 100%;
  z-index: 9;
`;

export const Link = styled.a`
  cursor: pointer;
  font-weight: ${(props) => props.weight};
  height: 100%;
  display: flex;
  align-items: center;
  text-transform: capitalize;
  color: ${(props) => props.theme.colors.white};
  font-size: ${(props) => props.theme.fontSizes.topMenu};
`;
export const Logo = styled.img`
  display: block;
  max-width: 100%;
  height: 4rem;
`;

export const Ul = styled.ul`
  display: flex;
  height: 100%;
  list-style: none;
  padding: 0;
`;

export const Li = styled.li`
  margin: 0 4px;
  height: 95%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 10px;
  &:hover {
    border-bottom: 2px solid #fff;
  }
`;
