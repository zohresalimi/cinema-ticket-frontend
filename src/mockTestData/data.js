import React, { useReducer } from "react";
import { Router } from "@reach/router";
import AppContext from "../store/context";
import Theme from "../Styles/Theme";
import reducer from "../store/reducer";

export const getTestStore = () => {
  return {
    user: {
      name: "",
      email: "",
    },
    movies: {
      children: [
        {
          _id: "5fa71823adc463397a50c501",
          actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
          age: "7",
          coverImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
          description:
            "Poppy and Kvist embark on an adventure that will take them far beyond",
          director: "aaron woodley",
          duration: "1:31",
          genre: ["family", "children"],
          images: [],
          largeImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
          name: "trolls 2: world tour",
          originalTitle: "swedish",
          premiere: "2020-10-03T15:18:52.722Z",
          rooms: ["5fa653d0a7f1201999dd1ac0"],
          trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
        },
      ],
      premiered: [
        {
          _id: "5fa71823adc463397a50c502",
          actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
          age: "7",
          coverImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
          description:
            "Poppy and Kvist embark on an adventure that will take them far beyond",
          director: "aaron woodley",
          duration: "1:31",
          genre: ["family", "children"],
          images: [],
          largeImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
          name: "follow me",
          originalTitle: "swedish",
          premiere: "2020-10-05T15:18:52.722Z",
          rooms: ["5fa653c0643c06586bba7d78"],
          trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
        },
      ],
      upcoming: [
        {
          _id: "5fa71823adc463397a50c503",
          actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
          age: "7",
          coverImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/68bfeef55ada4e7ab0606db2af0f40ad.jpg?width=240&version=EA2E80EE161E5DA161C6E26487054C9F&format=webp",
          description:
            "Poppy and Kvist embark on an adventure that will take them far beyond",
          director: "aaron woodley",
          duration: "1:31",
          genre: ["family", "children"],
          images: [],
          largeImage:
            "https://catalog.cinema-api.com/cf/images/ncg-images/79276cead82341d2a004f8e8f99a88e9.jpg?width=1920&version=08F0C83177D59FB94C4C809F63F67E2A&format=webp",
          name: "honest thief",
          originalTitle: "swedish",
          premiere: "2021-01-03T15:18:52.722Z",
          rooms: ["5fa653c0643c06586bba7d78"],
          trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
        },
      ],
      allMovies: {
        "5fa71823adc463397a50c501": "children",
        "5fa71823adc463397a50c502": "premiered",
        "5fa71823adc463397a50c503": "upcoming",
      },
    },
    cinemas: [
      {
        _id: "5fa64fe234fbbf9945f3a3ce",
        name: "filmstaden kista",
        purchaseEndTime: "21:30",
        purchaseStartTime: "8:00",
        rooms: ["5fa653c0643c06586bba7d78", "5fa653d0a7f1201999dd1ac0"],
      },
    ],
    ticket: {
      movie: {},
      status: false,
      quantity: 0,
      price: 0,
      seatNumbers: new Map(),
      user: {},
      showing: {},
    },
    showings: [
      {
        _id: "5fbe9bd31f58462b5c759b5c",
        capacity: 4,
        cinema: "5fa64fe234fbbf9945f3a3ce",
        endTime: null,
        movie: {
          _id: "5fa71823adc463397a50c502",
          actors: ["Holland Roden", "Ronen Rubinstein", "Keegan Allen"],
          age: "7",
          description:
            "Poppy and Kvist embark on an adventure that will take them far beyond",
          director: "aaron woodley",
          duration: "1:31",
          genre: ["family", "children"],
          images: [],
          name: "follow me",
          originalTitle: "swedish",
          premiere: "2020-10-05T15:18:52.722Z",
          rooms: ["5fa653c0643c06586bba7d78"],
          trailerUrl: "https://youtu.be/P9A6hOg9QQ0",
        },
        price: 100,
        room: { _id: "5fa653c0643c06586bba7d78" },

        seats: [
          [
            { seatNumber: 1, taken: "available" },
            { seatNumber: 2, taken: "available" },
          ],
          [
            { seatNumber: 1, taken: "available" },
            { seatNumber: 2, taken: "available" },
          ],
        ],
        startTime: "2020-11-13T16:00:00.000Z",
      },
    ],
  };
};

export const WithProvider = (props) => {
  const [state, dispatch] = useReducer(reducer, getTestStore());

  return (
    <Theme>
      <AppContext.Provider
        value={{
          state: props.defaultValue || state,
          dispatch: props.mockDispatch || dispatch,
        }}
      >
        <Router>{props.children}</Router>
      </AppContext.Provider>
    </Theme>
  );
};
