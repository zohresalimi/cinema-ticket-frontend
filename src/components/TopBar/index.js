import React, { useState, useRef } from "react";
import { useTranslation } from "react-i18next";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import DropDown from "../DropDown";
import HamburgerMenu from "../HamburgerMenu";
import { Wrapper, TopMenuContainer, Link, Logo, Ul, Li, Div } from "./style";
import { Container, Row, Col } from "../../Styles/StyleComponents";

function TopBar() {
  const [isMenuOpen, setMenuState] = useState(false);
  const node = useRef();
  const { t } = useTranslation();

  return (
    <Wrapper>
      <TopMenuContainer>
        <Container>
          <Row>
            <Col>
              <Div>
                <div ref={node}>
                  <HamburgerMenu
                    isOpen={isMenuOpen}
                    toggleMenu={setMenuState}
                  />
                </div>
                <Link href="/">
                  <Logo
                    alt=""
                    src="https://www.filmstaden.se/contentassets/abcfcecd76ac47a18718257ddc52e804/filmstadenloggo100ar.png"
                  />
                </Link>
                <Ul>
                  <Li>
                    <Link href="/" data-testid="home-item" weight="bold">
                      {t("home")}
                    </Link>
                  </Li>
                </Ul>
              </Div>
            </Col>
            <Col>
              <div className="searchBar">
                <Link href="/search" weight="bold" data-testid="serach-icon">
                  <FontAwesomeIcon icon={faSearch} className="searchIcon" />
                </Link>
                <DropDown />
              </div>
            </Col>
          </Row>
        </Container>
      </TopMenuContainer>
    </Wrapper>
  );
}

export default TopBar;
