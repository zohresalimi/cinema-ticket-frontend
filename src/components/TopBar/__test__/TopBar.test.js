import React from "react";
import { render, act } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import TopBar from "../";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (k) => k,
    i18n: {
      language: "sv",
      changeLanguage: () => new Promise(() => {}),
    },
  }),
}));

import { WithProvider } from "../../../mockTestData/data";

function renderWrapper() {
  let component;

  act(() => {
    component = render(
      <WithProvider>
        <TopBar path="/" />
      </WithProvider>
    );
  });

  return component;
}

describe("TopBar Component Testing", () => {
  test("take snapshot", async () => {
    const { container, getByTestId } = await renderWrapper();
    expect(getByTestId("home-item")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
  test("go to search page", async () => {
    const { getByTestId } = await renderWrapper();
    expect(getByTestId("serach-icon").getAttribute("href")).toBe("/search");
  });
});
