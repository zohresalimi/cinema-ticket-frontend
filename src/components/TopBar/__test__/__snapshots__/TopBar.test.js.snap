// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Component Testing take snapshot 1`] = `
<div
  style="outline: none;"
  tabindex="-1"
>
  <div
    class="style__Wrapper-sc-1ttkx56-2 giMqPh"
  >
    <div
      class="style__TopMenuContainer-sc-1ttkx56-1 gJqFou"
    >
      <div
        class="StyleComponents__Container-sc-11erdxt-0 fFgwWO"
      >
        <div
          class="StyleComponents__Row-sc-11erdxt-1 jhoSVF"
        >
          <div
            class="StyleComponents__Col-sc-11erdxt-2 dgPBuy"
          >
            <div
              class="style__Div-sc-1ttkx56-0 gFytCZ"
            >
              <div>
                <button
                  class="HamburgerMenu__StyledBurger-pqm95a-0 fuWDqq"
                  data-testid="burger-wrapper"
                >
                  <div />
                  <div />
                  <div />
                </button>
                <nav
                  class="HamburgerMenu__StyledMenu-pqm95a-1 dMjmXU"
                >
                  <a
                    href="/"
                  >
                    home
                  </a>
                  <a
                    href="/"
                  >
                    movies
                  </a>
                </nav>
              </div>
              <a
                class="style__Link-sc-1ttkx56-3 jcTWMg"
                href="/"
              >
                <img
                  alt=""
                  class="style__Logo-sc-1ttkx56-4 gqkUiH"
                  src="https://www.filmstaden.se/contentassets/abcfcecd76ac47a18718257ddc52e804/filmstadenloggo100ar.png"
                />
              </a>
              <ul
                class="style__Ul-sc-1ttkx56-5 hSnmco"
              >
                <li
                  class="style__Li-sc-1ttkx56-6 fMDjQV"
                >
                  <a
                    class="style__Link-sc-1ttkx56-3 dEHYxV"
                    data-testid="home-item"
                    href="/"
                  >
                    home
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="StyleComponents__Col-sc-11erdxt-2 dgPBuy"
          >
            <div
              class="searchBar"
            >
              <a
                class="style__Link-sc-1ttkx56-3 dEHYxV"
                data-testid="serach-icon"
                href="/search"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-search fa-w-16 searchIcon"
                  data-icon="search"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    fill="currentColor"
                  />
                </svg>
              </a>
              <div
                class="style__Wrapper-sc-13kue4j-0 bvCERe"
              >
                <button
                  class="style__Button-sc-13kue4j-3 dizWDM"
                  data-testid="toggle-dropdown"
                >
                  <p>
                    Svenska
                  </p>
                  <div
                    class="style__CaratContainer-sc-13kue4j-4 ljQyWE"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-angle-up fa-w-10 "
                      data-icon="angle-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </button>
                <ul
                  class="style__MenuLabel-sc-13kue4j-1 erqbDH"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
